### PR TITLE
Updates for dynamic token management

### PR DESCRIPTION
**Pull request by johnsmartco**

### CODE REVIEW

**Comment by rogerb-ds** [comment]
@johnsmartco It would be good to mention the ability to revoke a token/lease (vault lease revoke astra/org/token/<LEASE_ID>). I'm not sure of the best way to fit that in with the "delete" information. It may even replace it since with dynamic tokens they will always expire at some point. Deletion may no longer be necessary.

**Review by rogerb-ds — CHANGES_REQUESTED** [review]
Please include details on "vault lease revoke astra/org/token/<LEASE_ID>"

**Review by mendonk — APPROVED** [review]
Astra DB -> AstraDB
Other than that looks good to me.

File: docs/index.md [hunk 1]
```diff
@@ -271,34 +291,68 @@ Follow these steps:
...
-
+	
 	The command output displays the new token's properties, including:
 
 	* `clientId` (you'll see this value in Astra DB console too)
 	* `generatedOn` (date format)
 	* `metadata` (example: `map[purpose=demo user=mrsmart]`)
 	* `orgId` (its value)
 	* `token` (example: its `AstraCS:<generated-token-id>` value)
+	* `lease_time` (its value)
```
Comment (bansri29) [inline docs/index.md:324]: output also includes logical_name

File: docs/index.md [hunk 2]
```diff
@@ -271,34 +291,68 @@ Follow these steps:
...
+	vault list sys/leases/lookup/astra/org/token
+	```
 
-6. You can also delete tokens. Example:
+11. Delete tokens based on an Astra DB `org_id` and `role_name`. Example:
 
 	```bash
-	vault delete astra/org/token org_id="ccd999999_facd_4ad3_bbb99903d999999999999999d" role_name="organization_administrator"
+	vault delete astra/org/token org_id="ccd999999_facd_4ad3_bbb99903d999999999999999d" \
+	role_name="organization_administrator"
```
Comment (bansri29) [inline docs/index.md:355]: logical_name is required to delete a token.

**Review by bansri29 — COMMENTED** [review]
Ability to revoke a token before lease expires can be added. 
Also, logical_name is a required field to generate or delete a token.

File: docs/index.md [hunk 3]
```diff
@@ -14,11 +14,20 @@ Consequently, there's no audit trail of who has downloaded and used tokens, and
...
 
+Astra DB Plugin for HashiCorp Vault also gives you the ability to create and manage dynamic tokens outside of Astra Portal, which can then be used by your client applications that access Astra DB data. Using `vault` commands, authorized administrators can rotate tokens based on a token's lifetime lease. Specifically, you can: 
+
+* Define a default lease time
+* Create new tokens with lease settings
+* List tokens by each one's Client ID
+* Renew a lease
+* View lease details
+* List all leases
+
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:25]: Done in next commit.

File: docs/index.md [hunk 4]
```diff
@@ -271,34 +291,68 @@ Follow these steps:
...
+	role_name=Admin_Svc_Acct logical_name="My new token" lease_time=10m 
 	```
 
-	**TIP:** You can also apply custom, meaningful metadata to the generated Astra DB token by adding one or more `metadata` parameters. The metadata names and values can be any free-form text that you want. Example:
+	**TIP:** You can apply custom meaningful metadata to the generated Astra DB token by adding one or more `metadata` parameters. The metadata names and values can be any free-form text that you want. Here we'll also specify a lease_time. Example:
 
 	```bash
 	vault write astra/org/token org_id="ccd999999_facd_4ad3_bbb99903d999999999999999d" role_name="organization_administrator" \
-	 metadata="user=mrsmart" metadata="purpose=demo"
+	metadata="user=mrsmart" metadata="purpose=demo" lease_time=20m
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:314]: Hi @bansri29 , a late reply from me. What's an example of a logical_name=" "  value that would fit into this example ^^^ ?

[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:314]: Oh, actually from a prior example, something like:

vault write astra/config \ 
org_id=astra_token=AstraCS:YlABLSDOEMpQlrdWoLLJyzAh:8e34d55b6d774a7822ad87df2e502774749dc5549fd7e2bd248af307bee4ca8b \
 url=https://api.astra.datastax.com **logical_name=org_logical_name** renewal_time="4h"

[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:314]: Question: when are quotes needed?  Only when the value has a space?   We should renewal_time="4h" ... 

Should we always show quotes?  cc: @bansri29

[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:314]: I'll ask in Slack...

[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:314]: Updated with next commit.

File: docs/index.md (hunk 1 above)
Comment (johnsmartco) [reply docs/index.md:324]: Updated with next commit.

File: docs/index.md (hunk 2 above)
Comment (johnsmartco) [reply docs/index.md:355]: Thanks again - updated in next commit.

File: docs/index.md [hunk 5]
```diff
@@ -271,34 +291,68 @@ Follow these steps:
...
+
+	```bash
+	vault lease lookup astra/org/token/UuS2JYK9k5Di9dW9Zq4Ip3v1
+	```
+	
+10. List all leases. Example:
+
+	```bash
+	vault list sys/leases/lookup/astra/org/token
+	```
```
[the remark this answers is not in the document]
Comment (johnsmartco) [reply docs/index.md:361]: Done.

**Comment by johnsmartco** [comment]
Will resolve merge conflicts after next two meetings...

File: docs/index.md [hunk 6]
```diff
@@ -271,34 +295,76 @@ Follow these steps:
 	```
 
 	Also available is the `vault delete astra/role org_id="<ORG ID>" role="<ROLE NAME>"` command.
+	
+5. Define a default lease time. Example:
+
+	```bash
+	vault write astra/config \
+	org_id="astra_token=AstraCS:YlABLSDOEMpQlrdWoLLJyzAh:8e34d55b6d774a7822ad87df2e502774749dc5549fd7e2bd248af307bee4ca8b" \
+	url=:https://api.astra.datastax.com" logical_name="internalOrg" renewal_time="4h"
```
Comment (bansri29) [inline docs/index.md:304]: url="https://api.astra.datastax.com" instead of url=:https://api.astra.datastax.com"

**Review by bansri29 — APPROVED** [review]

File: docs/index.md (hunk 6 above)
Comment (johnsmartco) [reply docs/index.md]: Good catch - thanks Bansri.

**Comment by johnsmartco** [comment]
Thanks @bansri29. Just a note that I'll wait for a "go" signal from @rogerb-ds to merge the updates.  Probably this Thursday, 10-Nov.

**Review by rogerb-ds — APPROVED** [review]

**Pull request merged**